src/hotspot/os/bsd/os_perf_bsd.cpp

```diff
@@ -132,8 +132,12 @@ size_t length = sizeof(cpu_load_info);
     memcpy(cpu_load_info, &allcpus[which_logical_cpu * CPUSTATES], sizeof(long) * CPUSTATES);
     FREE_C_HEAP_ARRAY(long, allcpus);
 #else
-    /* TODO: NetBSD */
-    return FUNCTIONALITY_NOT_IMPLEMENTED;
+    char mib[18];
+    snprintf(mib, sizeof(mib), "kern.cp_time.%d", which_logical_cpu);
+    mib[sizeof(mib)-1] = 0;
```

Comment:
snprintf guarantees null termination even if buffer is too small for the result so `mib[sizeof(mib)-1] = 0;` is not needed.

While it is extremely unlikely that a system will have more then 9999 logical cpus's, we should probably not impose a limit here. mib array of at least 24 would be better I think (13 chars for 'kern.cp_time.' + 10 chars for max int value + 1 for NULL).

Reply:
Yes I wanted to sound "reasonable" in term of possible cpu support on NetBSD :-) but I can change no worries.